### PR TITLE
API Updates

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Fix a handful of documentation issues with the API. We're not ready to re-enable vetting of the OpenAPI spec, but this gets us (slightly) closer.

#### Ticket Link
None.

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: api/v4/source/files.yaml [hunk 1]
```diff
@@ -380,7 +380,7 @@
             fileID := "4xp9fdt77pncbef59f4k1qe83o"
 
             info, resp := Client.GetFileInfo(fileID)
-  "/api/v4/files/{file_id}/public":
+  "/files/{file_id}/public":
```
Comment (lieut-data) [inline api/v4/source/files.yaml:383]: This one git `/api/v4` added incorrectly -- fixing!

File: api/v4/source/groups.yaml [hunk 2]
```diff
@@ -275,6 +275,43 @@
           $ref: "#/components/responses/Forbidden"
         "501":
           $ref: "#/components/responses/NotImplemented"
+  "/api/v4/groups/{group_id}/restore":
```
Comment (lieut-data) [inline api/v4/source/groups.yaml:278]: Document one of the new APIs.

File: server/channels/api4/group.go [hunk 3]
```diff
@@ -84,7 +84,7 @@ func (api *API) InitGroup() {
 	api.BaseRoutes.Groups.Handle("/{group_id:[A-Za-z0-9]+}",
 		api.APISessionRequired(deleteGroup)).Methods("DELETE")
 
-	// GET /api/v4/groups/:group_id
+	// POST /api/v4/groups/:group_id
```
Comment (lieut-data) [inline server/channels/api4/group.go:87]: Wrong comment. Not sure this comment actually adds value, but fixing anyway.

**Review by lieut-data — COMMENTED** [review]
As an aside, I'm looking at alternative options for generating this documentation, since after some early review, it's /very/ stale and broken in ways that's nearly impossible to keep up-to-date. More on that later, but a few hand fixes for now.

**Review by hanzei — APPROVED** [review]
Thanks for keeping the API docs up to date (or at least trying to :wink: )

**Pull request merged**